packages/installer/src/executors/run-command-executor.tsx

```diff
@@ -0,0 +1,121 @@
+import {spawn} from "cross-spawn"
+import {Box, Text} from "ink"
+import Spinner from "ink-spinner"
+import * as React from "react"
+import {Newline} from "../components/newline"
+import {RecipeCLIArgs} from "../types"
+import {useEnterToContinue} from "../utils/use-enter-to-continue"
+import {useUserInput} from "../utils/use-user-input"
+import {Executor, executorArgument, ExecutorConfig, getExecutorArgument} from "./executor"
+
+export interface CliCommand {
+  command: string
+  commandArgs?: string[]
+}
+
+export interface Config extends ExecutorConfig {
+  command: executorArgument<CliCommand>
+}
+
+export interface CommitChildProps {
+  commandInstalled: boolean
+  handleChangeCommitted: () => void
+  step: Config
+  cliArgs: RecipeCLIArgs
+}
+
+export const type = "run-command"
+
+function Command({step, loading}: {step: Config; loading: boolean}) {
+  return (
+    <Text>
+      {`   `}
+      {loading ? <Spinner /> : "✅"}
+      {` ${step.command}`}
+    </Text>
+  )
+}
+
+const CommandList = ({
+  lede = "Hang tight! Running...",
+  commandLoading = false,
+  step,
+}: {
+  lede?: string
+  commandLoading?: boolean
+  step: Config
+}) => {
+  return (
+    <Box flexDirection="column">
+      <Text>{lede}</Text>
+      <Newline />
+      <Command key={step.stepId} step={step} loading={commandLoading} />
+    </Box>
+  )
+}
+
+/**
+ * Exported for unit testing purposes
+ */
+export async function executeCommand(command: string, commandArgs?: string[]) {
+  await new Promise((resolve) => {
+    const cp = spawn(command, commandArgs || [], {
+      stdio: ["inherit", "pipe", "pipe"],
+    })
+    cp.on("exit", resolve)
+  })
+}
+
+export const Commit: Executor["Commit"] = ({cliArgs, cliFlags, step, onChangeCommitted}) => {
+  const userInput = useUserInput(cliFlags)
+  const [commandInstalled, setCommandInstalled] = React.useState(false)
+
+  const handleChangeCommitted = React.useCallback(() => {
+    onChangeCommitted(`Executed command`)
+  }, [onChangeCommitted])
+
+  React.useEffect(() => {
+    async function runCommand() {
+      const executorCommand = getExecutorArgument((step as Config).command, cliArgs)
+      const command = executorCommand.command
+      const commandArgs = executorCommand.commandArgs
+      await executeCommand(command, commandArgs)
+      setCommandInstalled(true)
+    }
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    runCommand()
+  }, [cliArgs, step])
+
+  React.useEffect(() => {
+    if (commandInstalled) {
+      handleChangeCommitted()
+    }
+  }, [commandInstalled, handleChangeCommitted])
+
+  const childProps: CommitChildProps = {
+    commandInstalled,
+    handleChangeCommitted,
+    step,
```

Comment:
Not sure why, but currently the `step` definitions refers to `ExecutorConfig` instead of `Config`.
I checked the code from `add-dependency-executor.tsx`, but can't find a difference in the interface definition.

![image](https://user-images.githubusercontent.com/521777/143889024-eae2c5f2-8f75-4c21-83a4-469899d56051.png)

Reply:
Would removing `as Config` assertion from here: https://github.com/blitz-js/blitz/pull/3006/files/2d38191343de415a576bc00f35323bf1b40a837c..8947f0f1616ad19c93a86bed00cdd1951d36adc7#diff-7c110c48e211b614d1134a448d8200e1550607cee14aabee969f35d2e9e5cedeR79 make it work?

Reply:
Unfortunately not - If I remove `as Config`, `step` refers to `ExecutorConfig`

Reply:
Okay, got it. I guess you'll have to make an assertion for that (in line 98). It's not ideal, we could improve the types in the future.

Reply:
I think that's intended. Try passing the `command` through `proposalData` instead of `step`

Reply:
Hi,

thanks for the feedback.

> I think that's intended. Try passing the command through proposalData instead of step

To be honest, I have no glue how to do this 🙈 Maybe there is an example - but I wasn't able to find it, yet

I pushed some changes, which passes the command to execute as dedicated param.
Also I use now `step: step as Config`

Not sure if this is the best/correct way, but it solves the errors.